lib/percy/keyword_struct.rb

```diff
@@ -0,0 +1,21 @@
+# A struct that only allows keyword arguments.
```

Comment:
This code is extracted 1 for 1 and is not "new code" — everything else in this PR is technically "new" and should be reviewed.